autometa/taxonomy/ncbi.py

```diff
@@ -364,60 +347,68 @@ def parse_merged(self):
+            taxid_A_lineage_taxids))
+        common_ancestor.discard(1)  # This discards root
+        return True if common_ancestor else False
+
+    def is_valid_taxid(self, taxid):
+        """
+        Checks if the given taxid is a positive interger or not 
 
+        Parameters
+        ----------
+        taxid : int
+            identifer for a taxon in the Taxonomy Database by NCBI
 
-def main():
-    import argparse
-    import logging as logger
-    logger.basicConfig(
-        format='%(asctime)s : %(name)s : %(levelname)s : %(message)s',
-        datefmt='%m/%d/%Y %I:%M:%S %p',
-        level=logger.DEBUG)
-    parser = argparse.ArgumentParser(
-        description='Autometa NCBI utilities class')
-    parser.add_argument('ncbi', help='</path/to/ncbi/database/directory>')
-    parser.add_argument(
-        '--query-taxid',
-        help='query taxid to test NCBI class functionality',
-        default=1222,
-        type=int,
-    )
-    parser.add_argument('--verbose', help="add verbosity",
-                        action='store_true', default=True)
-    args = parser.parse_args()
-    ncbi = NCBI(args.ncbi, args.verbose)
-    query_taxid_parent = ncbi.parent(taxid=args.query_taxid)
-    logger.info(
-        f'{args.query_taxid} Name: {ncbi.name(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} Order: {ncbi.name(taxid=args.query_taxid, rank="order")}\n'
-        f'{args.query_taxid} Class: {ncbi.name(taxid=args.query_taxid, rank="class")}\n'
-        f'{args.query_taxid} Rank: {ncbi.rank(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} Lineage:\n{ncbi.lineage(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} is_common_ancestor {query_taxid_parent}: '
-        f'{ncbi.is_common_ancestor(parent_taxid=query_taxid_parent, child_taxid=args.query_taxid)}\n'
-    )
+        Returns
+        -------
+        boolean
+            True if the 'taxid' is a positive interger  else False
+        """
+
+        # This check if an interger has been added as str, eg. "562"
+        if isinstance(taxid, str):
+            if taxid.isnumeric() or taxid.replace('.', '', 1).isnumeric():
+                # checks if it is something like 12.0 vs. 12.9.
+                if float(taxid).is_integer():
+                    # See https://stackoverflow.com/a/47764450/12671809
+                    taxid = int(float(taxid))
+                else:
+                    return False
+            else:
+                return False
+        # `is_integer` returns error if only '12' is used, float(taxid) is needed
+        if not float(taxid).is_integer() or taxid < 0:
+            return False
+        return True
 
 
 if __name__ == '__main__':
-    main()
+    import argparse
+    parser = argparse.ArgumentParser(
+        description='file containing Autometa NCBI utilities class')
+    print('file containing Autometa NCBI utilities class')
+    args = parser.parse_args()
```

Comment:
Why do we have an argparse block here if we are not treating this script as a standalone module?

Reply:
I think `argparse block` will be required for the functionality of `parse_argaparse.py`

Reply:
`parse_argparse.py` is retrieving usage information for scripts that run as standalone modules. This script is not intended to be run as a standalone module, therefore it unnecessary to parse because it does not contain usage information.

Reply:
In that case the `parse_argparse.py` would return nothing which would seem a bit odd and out of place. 
Should I add `ncbi.py` and `utilities.py` as "exclude_scipts" in `parse_argparse.py`? It will be something similar to `exclude_dir` that we have currently. This would just skip those scripts rather than display a blank green box and help maintain uniformity.
https://github.com/KwanLab/Autometa/blob/ddc2bf951d6c4a8c29e9e2301e15a762327b0267/docs/parse_argparse.py#L327-L332

Reply:
No. returning nothing for a script that is _not_ a standalone module is what we want. This would create unnecessary extra maintenance.